src/sentry/api/endpoints/incident_details.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.base import Endpoint
+from sentry.api.bases.incident import IncidentPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.incidents.models import Incident
+from sentry.utils.sdk import configure_scope
+
+
+class IncidentDetailsEndpoint(Endpoint):
+    permission_classes = (IncidentPermission, )
+
+    def convert_args(self, request, incident_id, *args, **kwargs):
+        try:
+            incident = Incident.objects.select_related('organization').get(id=incident_id)
+        except Incident.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        self.check_object_permissions(request, incident.organization)
```

Comment:
Does this also check that the incident belongs to an organization you're a member of?

Reply:
It gets the organization related to the incident and checks you have permissions on that org. Or am I misunderstanding what you mean?